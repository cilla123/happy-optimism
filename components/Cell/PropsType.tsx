export default interface PropsType {
  theme?: 'default' | 'primary' | 'success' | 'warning' | 'error';
  hasArrow?: boolean;
  icon?: any;
  title?: any;
  description?: any;
  help?: any;
  disabled?: boolean;
  onClick?: () => void;
}
