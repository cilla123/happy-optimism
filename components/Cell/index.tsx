import React, { PureComponent } from 'react';
import PropsType from './PropsType';
import classnames from 'classnames';

export interface CellProps extends PropsType {
  prefixCls?: string;
  className?: string;
}

export default class Cell extends PureComponent<CellProps, {}> {

  static defaultProps = {
    prefixCls: 'za-cell',
    hasArrow: false,
    disabled: false,
  };

  render() {
    const {
      prefixCls,
      className,
      theme,
      hasArrow,
      icon,
      title,
      description,
      help,
      disabled,
      onClick,
      children,
      ...others,
    } = this.props;

    const cls = classnames(prefixCls, className, {
      [`theme-${theme}`]: !!theme,
      disabled,
      'is-link': !disabled && !!onClick,
      'has-icon': !!icon,
      'has-arrow': hasArrow,
    });

    const iconRender = icon && <div className={`${prefixCls}-icon`}>{icon}</div>;
    const titleRender = title && <div className={`${prefixCls}-title`}>{title}</div>;
    const contentRender = children && <div className={`${prefixCls}-content`}>{children}</div>;
    const arrowRender = hasArrow && <div className={`${prefixCls}-arrow`} />;
    const helpRender = help && (
      <div className={`${prefixCls}-help`}>
        {help}
      </div>
    );

    return (
      <div className={cls} onClick={onClick} onTouchStart={() => {}} {...others}>
        <div className={`${prefixCls}-inner`}>
          <div className={`${prefixCls}-header`}>
            {iconRender}
          </div>
          <div className={`${prefixCls}-body`}>
            {titleRender}
            {contentRender}
          </div>
          <div className={`${prefixCls}-footer`}>
            {description}
          </div>
          {arrowRender}
        </div>
        {helpRender}
      </div>
    );
  }
}
