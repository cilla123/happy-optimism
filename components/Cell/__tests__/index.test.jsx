import React from 'react';
import { render, shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Cell from '../index';

describe('Cell', () => {
  it('renders correctly', () => {
    const wrapper = render(<Cell title="标题文字" />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('description', () => {
    const wrapper = render(<Cell title="标题文字" description="描述文字" />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('icon', () => {
    const wrapper = render(<Cell title="标题文字" description="描述文字" icon={<img alt="" src="https://zhongantecheng.github.io/zarm/images/state.18e78939.png" />} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('children', () => {
    const wrapper = render(<Cell>this is children!</Cell>);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('help', () => {
    const wrapper = render(<Cell title="标题文字" description="描述文字" help={<span>message</span>} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('onClick', () => {
    const onClick = jest.fn();
    const wrapper = shallow(<Cell hasArrow title="标题文字" description="描述文字" onClick={onClick} />);
    expect(toJson(wrapper)).toMatchSnapshot();
    wrapper.simulate('click');
    expect(onClick).toBeCalled();
  });
});
