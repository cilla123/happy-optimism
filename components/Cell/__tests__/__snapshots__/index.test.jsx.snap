// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cell children 1`] = `
<div
  class="za-cell"
>
  <div
    class="za-cell-inner"
  >
    <div
      class="za-cell-header"
    />
    <div
      class="za-cell-body"
    >
      <div
        class="za-cell-content"
      >
        this is children!
      </div>
    </div>
    <div
      class="za-cell-footer"
    />
  </div>
</div>
`;

exports[`Cell description 1`] = `
<div
  class="za-cell"
>
  <div
    class="za-cell-inner"
  >
    <div
      class="za-cell-header"
    />
    <div
      class="za-cell-body"
    >
      <div
        class="za-cell-title"
      >
        标题文字
      </div>
    </div>
    <div
      class="za-cell-footer"
    >
      描述文字
    </div>
  </div>
</div>
`;

exports[`Cell help 1`] = `
<div
  class="za-cell"
>
  <div
    class="za-cell-inner"
  >
    <div
      class="za-cell-header"
    />
    <div
      class="za-cell-body"
    >
      <div
        class="za-cell-title"
      >
        标题文字
      </div>
    </div>
    <div
      class="za-cell-footer"
    >
      描述文字
    </div>
  </div>
  <div
    class="za-cell-help"
  >
    <span>
      message
    </span>
  </div>
</div>
`;

exports[`Cell icon 1`] = `
<div
  class="za-cell has-icon"
>
  <div
    class="za-cell-inner"
  >
    <div
      class="za-cell-header"
    >
      <div
        class="za-cell-icon"
      >
        <img
          alt=""
          src="https://zhongantecheng.github.io/zarm/images/state.18e78939.png"
        />
      </div>
    </div>
    <div
      class="za-cell-body"
    >
      <div
        class="za-cell-title"
      >
        标题文字
      </div>
    </div>
    <div
      class="za-cell-footer"
    >
      描述文字
    </div>
  </div>
</div>
`;

exports[`Cell onClick 1`] = `
<div
  className="za-cell is-link has-arrow"
  onClick={[Function]}
  onTouchStart={[Function]}
>
  <div
    className="za-cell-inner"
  >
    <div
      className="za-cell-header"
    />
    <div
      className="za-cell-body"
    >
      <div
        className="za-cell-title"
      >
        标题文字
      </div>
    </div>
    <div
      className="za-cell-footer"
    >
      描述文字
    </div>
    <div
      className="za-cell-arrow"
    />
  </div>
</div>
`;

exports[`Cell renders correctly 1`] = `
<div
  class="za-cell"
>
  <div
    class="za-cell-inner"
  >
    <div
      class="za-cell-header"
    />
    <div
      class="za-cell-body"
    >
      <div
        class="za-cell-title"
      >
        标题文字
      </div>
    </div>
    <div
      class="za-cell-footer"
    />
  </div>
</div>
`;
